plugins/pagerduty/src/api/client.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef } from '@backstage/core';
+import {
+  Service,
+  Incident,
+  RequestOptions,
+  ClientApiConfig,
+  ServicesResponse,
+  IncidentsResponse,
+  OnCallsResponse,
+  OnCall,
+} from '../components/types';
+import { PagerDutyClient, TriggerAlarmRequest } from './types';
+
+export class UnauthorizedError extends Error {}
+
+export const pagerDutyApiRef = createApiRef<PagerDutyClient>({
+  id: 'plugin.pagerduty.api',
+  description: 'Used to fetch data from PagerDuty API',
+});
+
+export class PagerDutyClientApi implements PagerDutyClient {
```

Comment:
I think there was a consensus on naming these the other way round. The interface would be `PagerDutyApi` amd the implementation would then be `PagerDutyClient`. @freben, @Rugvip - Can you confirm I'm correct here?

Reply:
I think that makes sense. That makes the ref `pagerDutyApiRef`